internal/reminder/metrics/provider.go

```diff
@@ -0,0 +1,132 @@
+package metrics
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/mindersec/minder/pkg/config/reminder"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/rs/zerolog/log"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+)
+
+const (
+	metricsPath       = "/metrics"
+	readHeaderTimeout = 2 * time.Second
+)
+
+// Provider manages the metrics server and OpenTelemetry setup
+type Provider struct {
+	server  *http.Server
+	mp      *sdkmetric.MeterProvider
+	metrics *Metrics
+}
+
+// NewProvider creates a new metrics provider
+func NewProvider(cfg *reminder.MetricsConfig) (*Provider, error) {
+	if cfg == nil {
+		return nil, errors.New("metrics config is nil")
+	}
+
+	if !cfg.Enabled {
+		return &Provider{}, nil
+	}
+
+	// Create Prometheus exporter
+	prometheusExporter, err := prometheus.New(
+		prometheus.WithNamespace("reminder_service"),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Prometheus exporter: %w", err)
+	}
+
+	// Create resource with service information
+	res := resource.NewWithAttributes(
+		semconv.SchemaURL,
+		semconv.ServiceName("reminder-service"),
+		semconv.ServiceVersion("v0.1.0"),
+	)
+
+	// Create meter provider
+	mp := sdkmetric.NewMeterProvider(
+		sdkmetric.WithReader(prometheusExporter),
+		sdkmetric.WithResource(res),
+	)
+
+	// Set global meter provider
+	otel.SetMeterProvider(mp)
+
+	// Create metrics
+	meter := mp.Meter("reminder-service")
+	metrics, err := NewMetrics(meter)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create metrics: %w", err)
+	}
```

Comment:
If you break the setup out into a static function, I'd probably have the calling reminder code look like:

```
if err := setupMetrics(ctx, cfg); err != nil {
  return err
}
reminder.metrics, err = NewMetrics(otel.Meter("reminder"))
if err != nil {
  return err
}
```

That separates the "system setup" of the metrics exporter from the provisioning of metrics for the main reminder loop, and makes it easier in the future to test the metrics without accidentally setting up a whole prometheus exporter.